madgraph/interface/reweight_interface.py

```diff
@@ -1024,58 +1313,258 @@ def change_kinematics(self, event):
+            self.set_initial_mass_to_zero(event_to_sud)
+            event_to_sud.check_kinematics_only()
+
+            gstr=event.aqcd*2.*pi
+            sorted_tag = (tuple(mapped_order[0]),tuple(sorted(mapped_order[1])))
+
+            # Read in the event momenta into np array in the order which is defined in the process directory
+            perm = []
+            i = 1
+            for r in mapped_order[1]:
+                if not list(sud_mod.original_pdg_list_dict[sorted_tag][1]).index(r) in perm:
+                    perm.append(list(sud_mod.original_pdg_list_dict[sorted_tag][1]).index(r))
+                else:
+                    perm.append(list(sud_mod.original_pdg_list_dict[sorted_tag][1]).index(r)+i)
+                    i += 1
+            order = dict((i,j) for i,j in enumerate(perm))
+            event_to_sud_order = copy.deepcopy(event_to_sud)
+            event_to_sud_order[:len(mapped_tag[0])] =  event_to_sud[:len(mapped_tag[0])]
+            for r in order.keys():
+                event_to_sud_order[r+len(mapped_tag[0])] = event_to_sud[order[r]+len(mapped_tag[0])]
+            p_in = np.zeros(shape=(n_part, 4))
+            for i,el in enumerate(event_to_sud_order):
+                p_in[i] = [float(el.E),float(el.px),float(el.py),float(el.pz)]
+            mapped_tag, mapped_order = event_to_sud_order.get_tag_and_order()
+            if list(sud_mod.original_pdg_list_dict[sorted_tag][1]) != mapped_order[1]:
+                logger.critical('ERROR: order in particle momenta does not match MG convention!')
+                sys.exit(3)
+
+            # compute the actual Sudakov weight
+            res = sud_mod.ewsudakov(sorted_tag, p_in, gstr)
+
+            # Do the reewightings
+            sudrat0 = 1. + res[1]/res[0] # SUD0            s_to_rij ON  rij_ge_mw ON
+            sudrat1 = 1. + res[2]/res[0] # SUD1 (SKD_weak) s_to_rij ON  rij_ge_mw ON
+            sudrat2 = 1. + res[3]/res[0] # SUD1 (SKD_weak) s_to_rij OFF  rij_ge_mw ON
+            sudrat3 = 1. + res[4]/res[0] # SUD1 (SKD_weak) s_to_rij OFF  rij_ge_mw OFF
+            sudrat4 = 1. + res[5]/res[0] # SUD1 (SKD_weak) s_to_rij ON  rij_ge_mw OFF
+
+            # Damp when the Sudakov weights are too large (in abs value)
+            large_sud_error=False
+            if abs(sudrat1) > 200:
+                logger.info('ERROR: event will not be reweighted because Sudakov ratio is too large: %s ' %sudrat1)
+                logger.info(buff_event)
+                sudrat0 = 1. 
+                sudrat1 = 1.
+                sudrat2 = 1.
+                sudrat3 = 1. 
+                sudrat4 = 1.
+                large_sud_error = True
+
+            # Dummy step: needed to read in the parse_reweight function
+            event.rescale_weights(1.)
+
+            rwgt_dict = copy.deepcopy(event.parse_reweight())
+            if rwgt_dict=={}:
+                rwgt_dict['1001'] =  orig_wgt
```

Comment:
CMS/ATLAS will likely not be happy with a name 1001... But not sure this is worth the change for the moment. (But I can bet they will ask)